github/commitstatus_test.go

```diff
@@ -113,6 +125,7 @@ func TestGitHubStatusSuccessMockAPI(t *testing.T) {
 					rateLimitReset:     now.Add(1 * time.Hour).Unix(),
 				},
 			},
+			wantSleeps: []time.Duration{0 * time.Second},
 		},
 		{
```

Comment:
I remember having a test when max attempts were reached. Can we add such a test here if it was removed?

Reply:
I have no recollection of having removed a test, although for sure I might have by mistake.

I will look into this and I also plan to make the tests more similar to prod (number of attempts = 3), because now they require to keep in mind that number of attempts = 2 to understand the failure logic. Now that the sleep is virtual, we can attempt = 3 as in prod.

Reply:
So I think this is the point. You added tests only to: `TestGitHubStatusSuccessMockAPI`
and you can catch this error only in `TestGitHubStatusFailureMockAPI` if you add do the same changes in that test and "measure" the time.

Reply:
Yes, I can now reproduce on that test indeed, thanks.

Reply:
Fixed.